packages/pipelines/src/hooks/useRuns.ts

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { differenceBy, uniqBy } from 'lodash-es';
+import {
+  K8sGroupVersionKind,
+  K8sResourceCommon,
+  getGroupVersionKindForModel,
+  useFlag,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { PipelineRunModel } from '../models';
+import { GetNextPage, Selector } from '../types/tekton-results';
+import { EQ } from '../utils/tekton-results-utils';
+import { FLAG_PIPELINE_TEKTON_RESULT_INSTALLED } from '../const';
+import { useDeepCompareMemoize } from '../utils/common-utils';
+import { useTRPipelineRuns } from './useTRPipelineRuns';
+import { useTRTaskRuns } from './useTRTaskRuns';
+
+export const useRuns = <Kind extends K8sResourceCommon>(
+  groupVersionKind: K8sGroupVersionKind,
+  namespace: string,
+  options?: {
+    selector?: Selector;
+    limit?: number;
+    name?: string;
+  },
+  cacheKey?: string,
+): [Kind[], boolean, unknown, GetNextPage] => {
+  const etcdRunsRef = React.useRef<Kind[]>([]);
+  const optionsMemo = useDeepCompareMemoize(options);
+  const isTektonResultEnabled = useFlag(FLAG_PIPELINE_TEKTON_RESULT_INSTALLED);
```

Comment:
`useFlag` hook resolves the feature is enabled or not in the host app (Console - which has feature flagging concept). This hook should not know anything about feature flag, pass it via parameters in the function so the host app will pass the value and this hook remains generic.

Reply:
Updated the PR.